spec/system/batch_edit_spec.rb

```diff
@@ -58,20 +66,43 @@
     expect(page).to have_selector("h1", text: "Admin Cart")
 
     # check changes were made, and non-expected changes were not made
+    work0.reload
     work1.reload
     work2.reload
 
+    # work0 is unchanged
+    expect(work0.provenance).to eq "provenance 0"
+    expect(work0.contained_by.map(&:title)).to eq []
+
+
     expect(work1.additional_title).to eq(["additional title 1a", "additional title 1b", "batch edit additional title"])
     expect(work1.provenance).to eq "batch edit provenance"
     expect(work1.creator).to eq([Work::Creator.new(category: "contributor", value: "creator 1")])
     expect(work1.description).to eq "description 1"
+    # Work 1 was already in collection 1: this batch edit should not change that.
+    expect(work1.contained_by.map(&:title)).to eq ["collection_1"]
 
     expect(work2.additional_title).to eq(["additional title 2a", "additional title 2b", "batch edit additional title"])
     expect(work2.provenance).to eq "batch edit provenance"
     expect(work2.creator).to eq([Work::Creator.new(category: "contributor", value: "creator 2")])
     expect(work2.description).to eq "description 2"
+    # Work 2 should have collection 1 added to it.
+    expect(work2.contained_by.first.title).to eq "collection_1"
+
+    # Go back to add collection 2 to both works:
```

Comment:
Adding another transaction in really slows down the test, makes it take longer.

Is there really anything we really need to do two batch edits (two submits) in order to test? 

Can't we just set up the initial data such that we can test everything we might want to test in only one batch edit operation? You have an initial condition of some records, you do a batch edit, test the post-edit condition to make sure it did all the things we wanted. Even adding more records if needed (although I'm not sure they are) will usually be faster/simpler than doing two batch edits. 

It's not clear to me what is being tested in this second batch edit -- didn't the first one already test adding a collection? If we want to test the ccondition of a record with a) no collections, b) a different collection, c) already had collection2 in it, can we just make three records with all three conditions for first batch edit?

Alternately, we could make a whole extra test case just for collection editing -- that's what "best practice" of test design would be... but it makes the test slower to do multiple requests like that. But packing multiple requests into just ONE test is the worst of both worlds...

Reply:
Fixed it!